flink-table/flink-table-code-splitter/src/main/java/org/apache/flink/table/codesplit/BlockStatementGrouper.java

```diff
@@ -0,0 +1,432 @@
+                String code, String context, long maxMethodLength, String parameters) {
+            this.tokenStream = new CommonTokenStream(new JavaLexer(CharStreams.fromString(code)));
+            this.rewriter = new TokenStreamRewriter(tokenStream);
+            this.context = context;
+            this.maxMethodLength = maxMethodLength;
+            this.parameters = parameters;
+        }
+
+        @Override
+        public Void visitStatement(StatementContext ctx) {
+
+            if (ctx.getChildCount() == 0) {
+                return null;
+            }
+
+            if (ctx.WHILE() != null || ctx.IF() != null || ctx.ELSE() != null) {
+                for (StatementContext statement : ctx.statement()) {
+                    if (shouldExtract(statement)) {
+                        groupBlock(statement);
+                    }
+                }
+            } else {
+                if (shouldExtract(ctx)) {
+                    groupBlock(ctx);
+                }
+            }
+
+            return null;
+        }
+
+        public List<Pair<String, List<LocalGroupElement>>> getGroups() {
+            List<Pair<String, List<LocalGroupElement>>> groupsTmp = new ArrayList<>();
+            for (BlockStatementGrouperVisitor child : children) {
+                groupsTmp.addAll(child.getGroups());
+            }
+
+            groupsTmp.addAll(this.groups);
+
+            return groupsTmp;
+        }
+
+        private void addGroups(List<Pair<String, List<LocalGroupElement>>> groups) {
+            this.groups.addAll(groups);
+        }
+
+        private void groupBlock(StatementContext ctx) {
+            int localCounter = this.counter++;
+            int localGroupCodeLength = 0;
+            List<LocalGroupElement> localGroup = new ArrayList<>();
+            for (BlockStatementContext bsc : ctx.block().blockStatement()) {
+
+                StatementContext statement = bsc.statement();
+                if (statement.IF() != null
+                        || statement.ELSE() != null
+                        || statement.WHILE() != null) {
+                    String localContext = context + "_rewriteGroup" + localCounter;
+                    BlockStatementGrouperVisitor visitor =
+                            new BlockStatementGrouperVisitor(
+                                    CodeSplitUtil.getContextString(statement),
+                                    localContext,
+                                    maxMethodLength,
+                                    parameters);
+                    JavaParser javaParser = new JavaParser(visitor.tokenStream);
+                    javaParser.getInterpreter().setPredictionMode(PredictionMode.SLL);
+                    visitor.visitStatement(javaParser.statement());
+
+                    localGroup.add(new RewriteContextGroupElement(statement, visitor.rewriter));
+                    children.add(visitor);
+                    localCounter = this.counter++;
```

Comment:
This block will be replaced by a method call. Add the length of method call to `localGroupCodeLength`?

Reply:
good catch, thanks. 
I've added method call to `localGroupCodeLength`.